setup/setup.sh

```diff
@@ -0,0 +1,183 @@
+}
+
+enable_remote_login() {
+    log "Enabling Remote Login (SSH)..."
+   
+    sudo systemsetup -setremotelogin on
+    
+    sudo launchctl load -w /System/Library/LaunchDaemons/ssh.plist 2>/dev/null || true
+    
+    log "Remote Login (SSH) enabled"
+}
+
+# Configure macOS updates
+configure_macos_updates() {
+    log "Configuring macOS updates for Download Only..."
+    
+    sudo defaults write /Library/Preferences/com.apple.SoftwareUpdate.plist AutomaticCheckEnabled -bool false
+    sudo defaults write /Library/Preferences/com.apple.SoftwareUpdate.plist AutomaticDownload -bool false
+    sudo defaults write /Library/Preferences/com.apple.SoftwareUpdate.plist AutomaticallyInstallMacOSUpdates -bool false
+    sudo defaults write /Library/Preferences/com.apple.SoftwareUpdate.plist CriticalUpdateInstall -bool false
+    sudo defaults write /Library/Preferences/com.apple.SoftwareUpdate.plist ConfigDataInstall -bool false
+    
+    sudo defaults write /Library/Preferences/com.apple.commerce.plist AutoUpdate -bool false
+    sudo defaults write /Library/Preferences/com.apple.commerce.plist AutoUpdateRestartRequired -bool false
+    
+    log "macOS updates configured for Download Only mode"
+}
+
+# Download and install Orka VM Tools
+install_orka_vm_tools() {
+    log "Installing Orka VM Tools version ${ORKA_VM_TOOLS_VERSION}..."
+    
+    local pkg_url="https://orka-tools.s3.amazonaws.com/orka-vm-tools/official/${ORKA_VM_TOOLS_VERSION}/orka-vm-tools.pkg"
+    local pkg_path="/tmp/orka-vm-tools.pkg"
+    local pkg_name="orka-vm-tools.pkg"
+    
+    log "Downloading Orka VM Tools from $pkg_url..."
+    curl -fsSL "$pkg_url" -o "$pkg_path" || error "Failed to download Orka VM Tools package"
+    
+    if [[ ! -f "$pkg_path" ]]; then
+        error "Orka VM Tools package not found after download"
+    fi
+    
+    log "Installing Orka VM Tools package..."
+    sudo installer -pkg "$pkg_path" -target / || error "Failed to install Orka VM Tools package"
+    
+    rm -f "$pkg_path"
+    
+    log "Orka VM Tools ${ORKA_VM_TOOLS_VERSION} installed successfully"
+}
+
+# Download and run the setup-sys-daemon.sh script
+setup_sys_daemon() {
+    log "Setting up system daemon..."
+    
+    local script_url="https://raw.githubusercontent.com/macstadium/packer-plugin-macstadium-orka/main/guest-scripts/setup-sys-daemon.sh"
+    
+    curl -fsSL "$script_url" | sudo bash || error "Failed to run setup-sys-daemon.sh"
+    
+    log "System daemon setup completed"
+}
+
+# Post-install system cleanup
+cleanup_system() {
+    log "Cleaning up system..."
+
+    osascript <<'EOF' 2>/dev/null || true
+tell application "System Events"
+    set quitapps to name of every application process whose background only is false
+end tell
+repeat with apps in quitapps
+    if apps is not in {"Finder", "loginwindow", "System Settings", "System Preferences"} then
+        try
+            tell application apps to quit
+        end try
+    end if
+end repeat
+EOF
```

Comment:
This does not work. It asks to kill Terminal - which also kills the running script (see the popup)

<img width="1920" height="1080" alt="Screenshot 2025-10-03 at 3 59 52 PM" src="https://github.com/user-attachments/assets/8f63799e-5fc0-4306-82ec-8b121b8a90bd" />

Do we really need any of the logic in this `cleanup_system` function? Note that removing from Downloads will also cause an alert to popup as you must grant Terminal access to this directory explicitly

I would opt to just keep it simple. And instruct users to just close open apps before pushing their image

Reply:
Is there a script in the Downloads folder now? Just clearing `/tmp` seems sufficient :)) Clearing the terminal history we can keep as well, but not mandatory.

In general - our script should not be too opinionated. The real goal is to
1. Make sure SSH / Screen Share are enabled
2. Install our VM tools
3. Run the launch daemon to improve screen sharing performance (not strictly required, but good to have)

Everything else is really up to the user how they wish to configure the guest OS - the above are what are required for the VM to work with Orka